src/main/java/io/jenkins/plugins/opentelemetry/job/MonitoringPipelineListener.java

```diff
@@ -209,26 +215,45 @@ private boolean isIgnoredStep(@Nullable StepDescriptor stepDescriptor) {
         return ignoreStep;
     }
 
-    private String getStepName(@Nullable StepDescriptor stepDescriptor, @Nonnull String name) {
+    private String getStepName(@Nonnull FlowNode node, @Nullable StepDescriptor stepDescriptor, @Nonnull String name) {
```

Comment:
why not removing the `stepDescriptor` parameter is is obtained doing a `node.getDescriptor()`?

Reply:
Unfortunately, some of the `onStart` have a different object that did not implement the `getDescriptor()` so I decided to pass the argument instead, but let me confirm it

Reply:
For the `getStepName` is possible but for the `getTypeName`, I'm afraid there are two type of consumers:

- `io.jenkins.plugins.opentelemetry.job.MonitoringPipelineListener#onStartNodeStep` with the  [`StepStartNode`](https://javadoc.jenkins.io/plugin/workflow-cps/org/jenkinsci/plugins/workflow/cps/nodes/StepStartNode.html) argument but it does implement `org.jenkinsci.plugins.workflow.cps.nodes.StepStartNode#getDescriptor`
- `io.jenkins.plugins.opentelemetry.job.MonitoringPipelineListener#onAtomicStep` with the [`StepAtomNode`](https://javadoc.jenkins.io/plugin/workflow-cps/org/jenkinsci/plugins/workflow/cps/nodes/StepAtomNode.html) argument that overrides `org.jenkinsci.plugins.workflow.graph.StepNode#getDescriptor`

I could not find any interface or super-class in common for `StepStartNode` and `StepAtomNode` , so I've decided to add a new parameter to the signature.

Reply:
👍